src/proto/h1/role.rs

```diff
@@ -289,13 +325,41 @@ impl Http1Transaction for Server {
         }))
     }
 
+    #[cfg_attr(feature = "layers",
+        instrument( level = "trace",
+                    skip(msg),
+                    fields( otel.name           = "Server::encode",
+                            otel.kind           = ?SpanKind::Server,
+                            otel.status_code    = ?opentelemetry::trace::StatusCode::Unset,
+                            otel.status_message = tracing::field::Empty,
+                            // OTel required (HTTP)
+                            http.host     = tracing::field::Empty,
+                            http.method   = tracing::field::Empty,
+                            http.scheme   = tracing::field::Empty,
+                            http.target   = tracing::field::Empty,
+                            http.url      = tracing::field::Empty,
+                            net.peer.ip   = tracing::field::Empty,
+                            net.peer.name = tracing::field::Empty,
+                            net.peer.port = tracing::field::Empty,
+                            // OTel optional (HTTP)
+                            http.flavor                 = tracing::field::Empty,
+                            http.request_content_length = tracing::field::Empty,
+                            // OTel optional (General)
+                            net.transport = "IP.TCP",
+                       )
+                    )
+                )]
     fn encode(mut msg: Encode<'_, Self::Outgoing>, dst: &mut Vec<u8>) -> crate::Result<Encoder> {
-        trace!(
-            "Server::encode status={:?}, body={:?}, req_method={:?}",
-            msg.head.subject,
-            msg.body,
-            msg.req_method
-        );
+        let mut otel_status_messages: Vec<String> = vec![];
+        if cfg!(feature = "layers") {
+            tracing::Span::current()
+                .record("http.method", &format!("{:?}", msg.req_method).as_str());
+            tracing::Span::current().record(
```

Comment:
it's probably more efficient to avoid getting the current span multiple times ---`Span::current` performs a thread-local storage access. prefer:
```suggestion
            let span = tracing::Span::current();
            span.record("http.method", &format!("{:?}", msg.req_method).as_str());
            span.record(
```

also, we should record the method as a `Debug` trait object, in order to avoid allocating a string here:
```suggestion
            let span = tracing::Span::current();
            span.record("http.method", &msg.req_method as &dyn std::fmt::Debug));
            span.record(
```

Reply:
> also, we should record the method as a Debug trait object, in order to avoid allocating a string here:

Thanks, that is way beyond my ken at this point.